### ISSUE TITLE
Any pages that require a barcode wont generate.

### ISSUE THREAD

**Issue by lawrencea1021** [jekkos/opensourcepos#97]
any page that generates a barcode from sales order to purchase order. or even just generating a bar code from inventory comes up  with a blank page. It writes to the sql. and has record just cant do anything but go back and it recreates the invoice. I just downloaded the software yesterday. Havent put it threw any real tests yet but will be soon. I promise to Donate once program fully operational. I also noticed when entering a customer name on the sales page and the customer doesnt exist (such as a typo) you get php errors

**Comment by odiea** [comment]
The only time I have php problems is if I try to use the current version of Xampp the previous version works much better. Don't know what you are using.

**Comment by jekkos** [comment]
Yes looks like you're getting some kind of php error there. Can you check the apache error_log and maybe add the error message?

Regarding the customer search, do you mean the search in the sale overview? I've just did a quick test there and I'm not seeing any errors after I input some random characters.

**Comment by lawrencea1021** [comment]
ok i am a little new to apache so i have corrected the barcode error .i tested it on other servers and didnt have the barcode issue. i was missing some parts in the core. so thats good. now i've tested this on multiple servers and i keep getting this error when i type in a customer that doesnt exist in sales mod.

A PHP Error was encountered
Severity: Notice
Message: Uninitialized string offset: 0
Filename: controllers/sales.php
Line Number: 439
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/paublic_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 110

I am still entry level php person but was trying to fix the error my self.

what i did is in sales.php line 439 i change 

$acronym .= $w[0];

to 

$acronym .= !empty($w);

and got NO errors at all but i still get a blank customer field and posts as an empty field

see attached picture

![image](https://cloud.githubusercontent.com/assets/13647937/9155073/dee6195a-3e79-11e5-89c3-f4ec2bcf8807.png)

**Comment by lawrencea1021** [comment]
I fixed the barcode issue however i keep getting the customer issue. The barcode issue was missing packet when installing apache. I have tested it on other server / shared hostings and didn't have the barcode issue so that was an error on my end. However everywhere i have installed i get the same php error with the customer.

A PHP Error was encountered
Severity: Notice
Message: Uninitialized string offset: 0
Filename: controllers/sales.php
Line Number: 439
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: libraries/Session.php
Line Number: 688
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 83
A PHP Error was encountered
Severity: Warning
Message: Cannot modify header information - headers already sent by (output started at /home/openpos/public_html/system/core/Exceptions.php:185)
Filename: controllers/secure_area.php
Line Number: 110 

I am still new to php but i tried fixing the error and on sales.php i edited line 439 

$acronym .= $w[0];

to

$acronym .= !empty($w);

and receive NO error but still blank customer and post as empty field

![image](https://cloud.githubusercontent.com/assets/13647937/9155119/4288b368-3e7b-11e5-823e-8e5e9669db56.png)
